lib/taskana-core-test/src/test/java/acceptance/taskcomment/query/QueryTaskCommentAccTest.java

```diff
@@ -0,0 +1,460 @@
+    userWithName.setFirstName("Max");
+    userWithName.setLastName("Mustermann");
+    userWithName.setFullName("Max Mustermann");
+    userService.createUser(userWithName);
+
+    defaultClassification =
+        DefaultTestEntities.defaultTestClassification().buildAndStore(classificationService);
+    defaultWorkbasket =
+        DefaultTestEntities.defaultTestWorkbasket().buildAndStore(workbasketService);
+    WorkbasketAccessItemBuilder.newWorkbasketAccessItem()
+        .workbasketId(defaultWorkbasket.getId())
+        .accessId(ACCESS_USER_1_1_ID)
+        .permission(WorkbasketPermission.OPEN)
+        .permission(WorkbasketPermission.READ)
+        .permission(WorkbasketPermission.READTASKS)
+        .permission(WorkbasketPermission.APPEND)
+        .buildAndStore(workbasketService);
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  class FilterTaskComments {
+
+    @WithAccessId(user = ACCESS_USER_1_1_ID)
+    @Test
+    void should_FilterTaskComments_For_IdIn() {
+      List<TaskComment> comments =
+          taskService.createTaskCommentQuery().idIn(comment1.getId(), comment2.getId()).list();
+      assertThat(comments).hasSize(2);
+    }
+
+    @WithAccessId(user = ACCESS_USER_1_1_ID)
+    @Test
+    void should_FilterTaskComments_For_TaskIdIn() {
+      List<TaskComment> comments =
+          taskService.createTaskCommentQuery().taskIdIn(task1.getId()).list();
+      assertThat(comments).hasSize(3);
+    }
+
+    @WithAccessId(user = "user-1-2")
+    @Test
+    void should_ThrowException_When_NotAuthorizedToReadTaskComments() {
+
+      ThrowingCallable call =
+          () -> taskService.createTaskCommentQuery().taskIdIn(task1.getId()).list();
+
+      assertThatThrownBy(call).isInstanceOf(NotAuthorizedToQueryWorkbasketException.class);
+    }
+
+    @WithAccessId(user = ACCESS_USER_1_1_ID)
+    @Test
+    void should_FilterTaskComments_For_CreatorIn() {
+      List<TaskComment> comments =
+          taskService.createTaskCommentQuery().creatorIn(ACCESS_USER_1_1_ID).list();
+      assertThat(comments).hasSize(2);
+    }
+
+    @WithAccessId(user = ACCESS_USER_1_1_ID)
+    @Test
+    void should_FilterTaskComments_For_CreatedWithin() {
+      TimeInterval timeInterval = new TimeInterval(Instant.now().minusSeconds(5), Instant.now());
+      List<TaskComment> comments =
+          taskService.createTaskCommentQuery().createdWithin(timeInterval).list();
+      assertThat(comments).hasSize(3);
+    }
+
+    @WithAccessId(user = ACCESS_USER_1_1_ID)
+    @Test
+    void should_FilterTaskComments_For_ModifiedWithin() {
+      TimeInterval timeInterval = new TimeInterval(Instant.now().minusSeconds(10), Instant.now());
+      List<TaskComment> comments =
+          taskService.createTaskCommentQuery().modifiedWithin(timeInterval).list();
+      assertThat(comments).hasSize(3);
+    }
+
+    @WithAccessId(user = ACCESS_USER_1_1_ID)
+    @Test
+    void should_FilterTaskComments_For_CreatorNotIn() {
```

Comment:
Can you regroup the tests? Placing the test for ABC_In and ABC_NotIn as well as ABC_Within and ABC_NotWithin right below each other would make it more readable

Reply:
Yes, I have grouped them together